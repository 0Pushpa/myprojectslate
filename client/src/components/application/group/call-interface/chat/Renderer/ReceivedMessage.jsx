import { Avatar } from "@material-ui/core";
import React from "react";
import DateFormatter from "utils/DateFormatter";

const ReceivedMessage = ({ i, messages, message }) => {
  return (
    <div className="d-flex justify-content-start mb-4 received-msg" key={i}>
      <div className="img_cont_msg">
        {i === messages.length - 1 && (
          <Avatar
            alt={`${message?.by?.split("")[0].toUpperCase()}`}
            src={`/static/images/avatar/${message + 1}.jpg`}
          />
        )}
        {i < messages.length - 1
          ? messages.slice(0).reverse()[i + 1].by !== message.by && (
              <Avatar
                alt={`${message?.by?.split("")[0].toUpperCase()}`}
                src={`/static/images/avatar/${message + 1}.jpg`}
              />
            )
          : ""}
      </div>
      <div className="msg_cotainer">
        {i < messages.length - 1
          ? messages.slice(0).reverse()[i + 1].by !== message.by && (
              <p className="sent__by">{message.by}</p>
            )
          : ""}
        {message.message}
        <span className="msg_time">{DateFormatter(message.sent_at)}</span>
      </div>
    </div>
  );
};

export default ReceivedMessage;
