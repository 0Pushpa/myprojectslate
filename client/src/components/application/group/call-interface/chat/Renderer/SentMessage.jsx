import React from "react";
import DateFormatter from "utils/DateFormatter";

const SentMessage = (props) => {
  const sentRef = React.useRef();

  return (
    <div
      className="d-flex justify-content-end sent-msg"
      style={{ marginBottom: "1.6rem" }}
      key={props.i}
    >
      <div className="msg_cotainer_send" ref={sentRef}>
        {props.message.message}
        <span className="msg_time_send">
          {DateFormatter(props.message.sent_at)}
        </span>
      </div>
    </div>
  );
};

export default SentMessage;
