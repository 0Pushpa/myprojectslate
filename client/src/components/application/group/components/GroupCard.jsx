import { Box, Grid, IconButton } from "@material-ui/core";
import NameLogo from "components/elements/NameLogo";
import React from "react";
import { FiUserPlus, FiZap } from "react-icons/fi";
import { HiOutlineDotsHorizontal } from "react-icons/hi";
import { Link } from "react-router-dom";
import { EditGroupService } from "../../../../services/GroupService";
import CreateGroup from "../group-modal/CreateGroupModal";

const GroupCard = ({ index, ad, deleteGroup, getGroups }) => {
  const [anchorEl, setAnchorEl] = React.useState(null);
  const [showModal, setShowModal] = React.useState(false);
  const [groupName, setGroupName] = React.useState(ad?.GroupID?.name);

  const toggleModal = () => {
    setShowModal(!showModal);
    setAnchorEl(false);
  };
  const handleClick = () => {
    setAnchorEl(!anchorEl);
  };

  const editGroup = async () => {
    let data = {
      id: ad?.GroupID?._id,
      name: groupName,
    };
    const res = await EditGroupService(data);
    if (res?.status === 200) {
      toggleModal();
      getGroups();
    }
  };

  const handleChange = (e) => {
    setGroupName(e.target.value);
  };

  return (
    <Grid
      item
      xs={6}
      sm={4}
      md={3}
      lg={2}
      component="div"
      className=" group__box"
      title={ad.name}
      key={index}
      style={{ paddingBottom: "15px", position: "relative" }}
    >
      {/* <Box
        style={{
          position: "absolute",
          right: "20px",
          zIndex: "999",
        }}
        component="span"
        className="group-interface-dots"
        id="basic-button-2"
        aria-controls="basic-menu-2"
        aria-haspopup="true"
        aria-expanded={open ? "true" : undefined}
        onClick={handleClick}
      >
        <HiOutlineDotsHorizontal />
          </Box> */}
      {ad?.UserID === ad?.GroupID?.createdBy && (
        <IconButton
          style={{
            position: "absolute",
            right: "20px",
            zIndex: "999",
          }}
          onClick={handleClick}
        >
          <HiOutlineDotsHorizontal />
        </IconButton>
      )}

      {anchorEl && (
        <Box component="div" className="group-interface-dots-menu">
          <ul>
            <li onClick={toggleModal}>
              Edit Group
              <Box component="span">
                <FiZap />
              </Box>
            </li>
            <li onClick={() => deleteGroup(ad.GroupID._id)}>
              Delete Group
              <Box component="span" className="group-interface-add-user">
                <FiUserPlus />
              </Box>
            </li>
          </ul>
        </Box>
      )}
      <Link to={"/slate/groups/" + ad.GroupID._id}>
        <Box component="div" className="group-interface-firstgroup">
          <Box component="div" className="group-interface-imgcontainer">
            <NameLogo group={ad.GroupID.name} />
          </Box>
          <Box component="div" className="group-desc">
            <Box component="div" className="group-interface-text">
              <h3
                className="group-interface-h3"
                style={{
                  textTransform: "capitalize",
                }}
              >
                {" "}
                {ad.GroupID.name}
              </h3>
            </Box>
          </Box>
        </Box>
      </Link>
      <CreateGroup
        isEdit={true}
        show={showModal}
        handleClose={toggleModal}
        service={editGroup}
        groupName={groupName}
        handleChange={handleChange}
      />
    </Grid>
  );
};

export default GroupCard;
