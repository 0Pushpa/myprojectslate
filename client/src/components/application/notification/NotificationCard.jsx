import React, { useEffect } from "react";
import styled from "styled-components";
import { GrCircleAlert } from "react-icons/gr";
import Box from "@mui/material/Box";
import Button from "@mui/material/Button";
import Stack from "@mui/material/Stack";
import { Link } from "react-router-dom";
// import ErrorIcon from "@mui-material/icons/Error";
import DateFormatter from "../../../utils/DateFormatter";
import {
  AcceptNotificationsService,
  RejectNotificationsService,
} from "../../../services/NotificationService";
import { MainSocketContext } from "../Frontendmain";
import { ChatService } from "services/GroupService";
import { useHistory } from "react-router-dom";
import CircularProgress from "@material-ui/core/CircularProgress";

const H2 = styled.h3`
  font-size: 0.9rem;
  font-weight: lighter;
  display: inline-block;
  padding-left: 10px;
`;
const AlertIcon = styled(GrCircleAlert)`
  color: green;
`;
const MainDiv = styled.div`
  padding: 10px;
  border-radius: 5px;
  background-color: #f1f3f8;
  margin: 5px;
`;
const Buttons = styled(Button)`
  height: 30px;
  font-size: 0.8rem;
`;
const Time = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  justify-content: flex-end;
`;
const H1 = styled.h3`
  font-size: 0.7rem;
  font-weight: 300;
  color: #a3a3a3;
`;

const ActionMessage = styled.div`
  color: #575757;
  font-size: 0.8rem;
  padding: 5px 0;
  min-width: 200px;
`;

export default function Notification(props) {
  const [decision, setDecision] = React.useState(
    props.data.remark || "pending"
  );

  const [loading, setLoading] = React.useState(false);
  const [active, setActive] = React.useState(0);
  const history = useHistory();

  const context = React.useContext(MainSocketContext);

  const handleButtonClick = async (decision) => {
    setLoading(true);

    let res;
    const data = {
      notificationId: props.data._id,
      userId: props.data.UserID._id,
      groupId: props.data.url.split("/")[3],
    };

    if (decision === "accepted") {
      res = await AcceptNotificationsService(data);
    } else {
      res = await RejectNotificationsService(data);
    }
    if (res?.data?.status === "success") {
      setDecision(decision);

      if (decision === "accepted") {
        setLoading(false);
        const res = await ChatService({
          FromID: props.data.UserID._id,
          ToID: props.data.url.split("/")[3],
          userName: props.data.UserID.name,
          message: `${props.data.UserID.name} have joined the group`,
          messageType: "notification",
          status: true,
        });
        if (res?.status === 200) {
          context.socket.emit("joined alert in chat", {
            message: `${props.data.UserID.name} have joined the group`,
            by: props.data.UserID.name,
            uid: props.data.UserID._id,
            groupId: props.data.url.split("/")[3],
            sent_at: new Date().toLocaleString(),
            messageType: "notification",
          });
          history.push(props.data.url);
        }
      }
    }
  };
  return (
    <MainDiv component="div">
      <Box component="span" className="notification_svg">
        <AlertIcon />
      </Box>
      <H2>
        <div dangerouslySetInnerHTML={{ __html: props.data.message }}></div>
      </H2>
      {/* <Mention>{props.data.groupname}</Mention> */}
      <Box component="span" paddingLeft={2}>
        {props.data.icon}
      </Box>
      <Stack direction="row" spacing={2} marginLeft={2} marginTop={1}>
        {props.data.type === "confirmation" && decision === "pending" ? (
          <>
            <Buttons
              onClick={() => {
                setActive(0);
                handleButtonClick("accepted");
              }}
              variant="contained"
              style={{ background: "#00aaef" }}
            >
              {active === 0 && loading ? (
                <CircularProgress
                  style={{
                    width: "20px",
                    height: "20px",
                    color: "#fff",
                  }}
                />
              ) : (
                "Accept"
              )}
            </Buttons>

            <Buttons
              variant="outlined"
              color="error"
              onClick={() => {
                setActive(1);
                handleButtonClick("rejected");
              }}
            >
              {active === 1 && loading ? (
                <CircularProgress
                  style={{
                    width: "20px",
                    height: "20px",
                    color: "#fff",
                  }}
                />
              ) : (
                "Reject"
              )}
            </Buttons>
          </>
        ) : (
          <>
            {decision === "accepted" ? (
              <ActionMessage>Invitation Accepted</ActionMessage>
            ) : (
              <ActionMessage>Invitation Rejected</ActionMessage>
            )}
          </>
        )}
        <Time component="div">
          <H1>{DateFormatter(props.data.createdAt)}</H1>
        </Time>
      </Stack>
    </MainDiv>
  );
}
