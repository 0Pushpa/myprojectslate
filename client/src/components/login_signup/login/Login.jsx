import Alert from "@material-ui/lab/Alert";
import React, { useState } from "react";
import { FaFacebookF, FaGoogle, FaInstagram } from "react-icons/fa";
import { store } from "react-notifications-component";
import { useDispatch } from "react-redux";
import { useHistory } from "react-router";
import { authData } from "../../../redux/action/Index";
import { LoginUserService } from "../../../services/AuthService";
import { Link } from "react-router-dom";
// import * as React from 'react';
// import Box from "@mui/material/Box";
// import { TextField } from "@material-ui/core";
const Login = () => {
  const [getUserEmail, setGetUserEmail] = useState("");
  const [getUserPassword, setGetUserPassword] = useState("");
  const [error, setError] = useState(false);
  // const [errorMessage, setErrorMessage] = useState("");

  const closeAlert = () => {
    setError(false);
  };

  const dispatch = useDispatch();
  const history = useHistory();
  const userEmail = (event) => {
    setGetUserEmail(event.target.value);
  };
  const userPassword = (event) => {
    setGetUserPassword(event.target.value);
  };
  async function send(e) {
    e.preventDefault();
    const payload = {
      email: getUserEmail,
      password: getUserPassword,
    };

    // const delay = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

    const res = await LoginUserService(payload);
    if (res && res.data?.status === "success") {
      localStorage.setItem("user-token", res.data.token);
      dispatch(authData(res.data.user));
      history.push("/slate");
      store.addNotification({
        title: "Success",
        message: "Logged In Successfully",
        type: "success",
        insert: "bottom",
        container: "bottom-right",
        animationIn: ["animate__animated", "animate__fadeIn"],
        animationOut: ["animate__animated", "animate__fadeOut"],
        dismiss: {
          duration: 3000,
          onScreen: true,
        },
      });
    } else {
      console.log("error occured", res);
      if (res.data.status === "unauthorized") {
        store.addNotification({
          title: "Notice",
          message: res.data.message,
          type: "info",
          insert: "top",
          container: "top-right",
          animationIn: ["animate__animated", "animate__fadeIn"],
          animationOut: ["animate__animated", "animate__fadeOut"],
          dismiss: {
            duration: 6000,
            onScreen: true,
          },
        });
      } else {
        store.addNotification({
          title: "Error Occurred",
          message: res.data.message,
          type: "danger",
          insert: "top",
          container: "top-right",
          animationIn: ["animate__animated", "animate__fadeIn"],
          animationOut: ["animate__animated", "animate__fadeOut"],
          dismiss: {
            duration: 6000,
            onScreen: true,
          },
        });
      }

      // setErrorMessage(res.message);
      console.log(res);
      dispatch({
        type: "LOGIN_ERROR",
        payload: res.message,
      });
    }
  }
  return (
    <div className="form-container sign-in-container">
      <form onSubmit={send} className="login__form">
        {error && (
          <Alert severity="error" onClose={closeAlert}>
            Invalid credentials provided
          </Alert>
        )}
        <h1>Login</h1>
        <div className="social-container">
          <a href="https://www.facebook.com" className="social">
            <FaFacebookF />
          </a>
          <a href="https://www.instagram.com" className="social">
            <FaInstagram />
          </a>
          <a href="https://www.google.com" className="social">
            <FaGoogle />
          </a>
        </div>
        <span>or use your account</span>
        <input
          type="email"
          placeholder="Email"
          value={getUserEmail}
          onChange={userEmail}
        />
        <input
          type="password"
          placeholder="Password"
          value={getUserPassword}
          onChange={userPassword}
        />

        <span className="forgot_password_text_login">
          <Link to="/forgot-password">Forgot your password?</Link>
        </span>
        <button className="signin">LogIn</button>
      </form>
    </div>
  );
};

export default Login;