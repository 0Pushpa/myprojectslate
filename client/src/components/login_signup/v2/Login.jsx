import React, { useState } from "react";
import { MdOutlineMail } from "react-icons/md";
import { useDispatch } from "react-redux";
import { useHistory } from "react-router";
import { Link } from "react-router-dom";
import { authData } from "../../../redux/action/Index";
import { LoginUserService } from "../../../services/AuthService";
import "../Login_Signup.css";
import PasswordWithToggle from "../../elements/PasswordWithToggle";
import { InputAdornment, TextField } from "@material-ui/core";
import { useForm } from "react-hook-form";
import { yupResolver } from "@hookform/resolvers/yup";
import * as Yup from "yup";

export default function Login({ openSignUp }) {
  const LoginSchema = Yup.object().shape({
    email: Yup.string()
      .email("Must be a valid email")
      .max(255)
      .required("Email is required"),
    password: Yup.string().required("Password is required"),
  });

  const {
    register,
    handleSubmit,
    setValue,
    formState: { errors },
  } = useForm({
    resolver: yupResolver(LoginSchema),
    defaultValues: {
      email: null,
      password: null,
    },
  });

  const [getUserPassword, setGetUserPassword] = useState("");

  const dispatch = useDispatch();
  const history = useHistory();

  const userPassword = (event) => {
    setValue("password", event.target.value);
  };

  async function send(payload) {
    const res = await LoginUserService(payload);
    if (res && res.data?.status === "success") {
      // Handle successful login
      localStorage.setItem("user-token", res.data.token);
      localStorage.setItem("refresh-token", res.data.refreshToken);
      dispatch(authData(res.data.user));
      history.push("/slate");
    } else {
      console.log("Error occurred", res);

      // Handle unauthorized or other errors
      if (res?.data?.status === "unauthorized") {
        console.log("Unauthorized:", res.data.message);
      } else {
        console.log("Error message:", res?.data?.message);
      }

      // Dispatch error action
      dispatch({
        type: "LOGIN_ERROR",
        payload: res.message,
      });
    }
  }

  return (
    <form className="sign-up-form" onSubmit={handleSubmit(send)}>
      <h2 className="title">Login</h2>
      <TextField
        fullWidth
        required
        style={{ maxWidth: "380px" }}
        id="email-input"
        label="Email"
        variant="outlined"
        autoFocus
        {...register("email")}
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              <MdOutlineMail />
            </InputAdornment>
          ),
        }}
        error={Boolean(errors.email)}
        helperText={errors.email && errors.email.message}
      />
      <PasswordWithToggle
        onType={userPassword}
        error={Boolean(errors.password)}
        helperText={errors.password && errors.password.message}
      />
      <input type="submit" value="Login" className="btn solid" />
      <p className="social-text">
        <Link style={{ color: "#0f9ce0" }} to="/forgot-password">
          Forgot Password?
        </Link>
      </p>
    </form>
  );
}