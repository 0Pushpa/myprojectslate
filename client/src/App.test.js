import React from "react";
import { shallow, mount } from "enzyme";
import "setupTests.js";
import { BrowserRouter as Router } from "react-router-dom";
//redux
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
//message
import SentMessage from "components/application/group/call-interface/chat/Renderer/SentMessage";
import ReceivedMessage from "components/application/group/call-interface/chat/Renderer/ReceivedMessage";
//notification
import NotificationCard from "components/application/notification/NotificationCard";
//group
import GroupCard from "components/application/group/components/GroupCard";
//auth
import Login from "components/login_signup/v2/Login";

describe("Authentication Components", () => {
  const initialState = { output: 10 };
  const mockStore = configureStore();
  let store;
  it("should create an entry in login state", () => {
    store = mockStore(initialState);
    const component = mount(
      <Router>
        <Provider store={store}>
          <Login />
        </Provider>
      </Router>
    );
    const form = component.find("#email-input").first();
    form.props().onChange({
      target: {
        name: "myName",
        value: "myValue",
      },
    });
    expect(0).toEqual(0);
  });
});

describe("Attendance Components", () => {
  const initialState = { output: 10 };
  const mockStore = configureStore();
  let store;
  it("should create an entry in login state", () => {
    store = mockStore(initialState);
    const component = mount(
      <Router>
        <Provider store={store}>
          <Login />
        </Provider>
      </Router>
    );
    const form = component.find("#email-input").first();
    form.props().onChange({
      target: {
        name: "myName",
        value: "myValue",
      },
    });
    expect(0).toEqual(0);
  });
});

describe("Assignment Components", () => {
  const initialState = { output: 10 };
  const mockStore = configureStore();
  let store;
  it("should create an entry in login state", () => {
    store = mockStore(initialState);
    const component = mount(
      <Router>
        <Provider store={store}>
          <Login />
        </Provider>
      </Router>
    );
    const form = component.find("#email-input").first();
    form.props().onChange({
      target: {
        name: "myName",
        value: "myValue",
      },
    });
    expect(0).toEqual(0);
  });
});

describe("Chat Components", () => {
  it("should render sent message", () => {
    const component = shallow(
      <SentMessage
        i="1"
        message={{
          message: "eje",
          by: "Aman Maharjan",
          uid: "61b704402e69c9b32d73c90f",
          groupId: "6237e780a159bc00abc9715f",
          sent_at: "2022-03-21T02:58:39.802Z",
          messageType: "message",
          type: "sent",
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("should render receiving message", () => {
    const component = shallow(
      <ReceivedMessage
        i="1"
        messages={[
          {
            message: "eje",
            by: "Aman Maharjan",
            uid: "61b704402e69c9b32d73c90f",
            groupId: "6237e780a159bc00abc9715f",
            sent_at: "2022-03-21T02:58:39.802Z",
            messageType: "message",
            type: "received",
          },
          {
            message: "ok",
            by: "Baman Maharjan",
            uid: "61b704402e69c9b32d73c94f",
            groupId: "6237e780a159bc00abc9715f",
            sent_at: "2022-04-21T02:58:39.802Z",
            messageType: "message",
            type: "received",
          },
        ]}
        message={{
          message: "eje",
          by: "Aman Maharjan",
          uid: "61b704402e69c9b32d73c90f",
          groupId: "6237e780a159bc00abc9715f",
          sent_at: "2022-03-21T02:58:39.802Z",
          messageType: "message",
          type: "received",
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
});

describe("Group Components", () => {
  it("should display group list", () => {
    const component = shallow(
      <GroupCard
        index="1"
        ad={{
          _id: "61ce780a0f14779f59d64588",
          GroupID: {
            _id: "61caa5206d4819a4121b77bd",
            name: "Aman Group",
            status: true,
            type: "private",
            createdBy: "61b707b5c25f987fb2af5594",
            createdAt: "2021-12-28T05:48:16.277Z",
            updatedAt: "2021-12-28T05:48:16.277Z",
          },
          UserID: "61b704402e69c9b32d73c90f",
          role: "user",
          status: true,
          createdAt: "2021-12-31T03:24:58.006Z",
          updatedAt: "2021-12-31T03:24:58.006Z",
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
});

describe("Notification Components", () => {
  it("should display notification list", () => {
    const component = shallow(
      <NotificationCard
        data={{
          _id: "61c9fe8d4c6bc8a0583aef7c",
          UserID: "61b707b5c25f987fb2af5594",
          message: "You have been invited to join a group",
          type: "confirmation",
          url: "/slate/groups/61b62b0b5677213522aad463",
          remark: "",
          status: false,
          createdAt: "2021-12-27T17:57:33.172+00:00",
          updatedAt: "2021-12-27T17:57:33.172+00:00",
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
