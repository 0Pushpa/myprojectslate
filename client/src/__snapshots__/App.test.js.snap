// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Components should render receiving message 1`] = `
<div
  className="d-flex justify-content-start mb-4 received-msg"
  key="1"
>
  <div
    className="img_cont_msg"
  />
  <div
    className="msg_cotainer"
  >
    eje
    <span
      className="msg_time"
    >
      Mar 21 08:03 AM
    </span>
  </div>
</div>
`;

exports[`Chat Components should render sent message 1`] = `
<div
  className="d-flex justify-content-end sent-msg"
  key="1"
  style={
    Object {
      "marginBottom": "1.6rem",
    }
  }
>
  <div
    className="msg_cotainer_send"
  >
    eje
    <span
      className="msg_time_send"
    >
      Mar 21 08:03 AM
    </span>
  </div>
</div>
`;

exports[`Group Components should display group list 1`] = `
<WithStyles(ForwardRef(Grid))
  className=" group__box"
  component="div"
  item={true}
  key="1"
  lg={2}
  md={3}
  sm={4}
  style={
    Object {
      "paddingBottom": "15px",
      "position": "relative",
    }
  }
  xs={6}
>
  <Link
    to="/slate/groups/61caa5206d4819a4121b77bd"
  >
    <Styled(MuiBox)
      className="group-interface-firstgroup"
      component="div"
    >
      <Styled(MuiBox)
        className="group-interface-imgcontainer"
        component="div"
      >
        <NameLogo
          group="Aman Group"
        />
      </Styled(MuiBox)>
      <Styled(MuiBox)
        className="group-desc"
        component="div"
      >
        <Styled(MuiBox)
          className="group-interface-text"
          component="div"
        >
          <h3
            className="group-interface-h3"
            style={
              Object {
                "textTransform": "capitalize",
              }
            }
          >
             
            Aman Group
          </h3>
        </Styled(MuiBox)>
      </Styled(MuiBox)>
    </Styled(MuiBox)>
  </Link>
  <GroupModal
    groupName="Aman Group"
    handleChange={[Function]}
    handleClose={[Function]}
    isEdit={true}
    service={[Function]}
    show={false}
  />
</WithStyles(ForwardRef(Grid))>
`;

exports[`Notification Components should display notification list 1`] = `
<styled.div
  component="div"
>
  <ForwardRef(Box)
    className="notification_svg"
    component="span"
  >
    <Styled(GrCircleAlert) />
  </ForwardRef(Box)>
  <styled.h3>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "You have been invited to join a group",
        }
      }
    />
  </styled.h3>
  <ForwardRef(Box)
    component="span"
    paddingLeft={2}
  />
  <ForwardRef(Stack)
    direction="row"
    marginLeft={2}
    marginTop={1}
    spacing={2}
  >
    <Styled(Component)
      onClick={[Function]}
      style={
        Object {
          "background": "#00aaef",
        }
      }
      variant="contained"
    >
      Accept
    </Styled(Component)>
    <Styled(Component)
      color="error"
      onClick={[Function]}
      variant="outlined"
    >
      Reject
    </Styled(Component)>
    <styled.div
      component="div"
    >
      <styled.h3>
        Dec 27 23:12 PM
      </styled.h3>
    </styled.div>
  </ForwardRef(Stack)>
</styled.div>
`;
